CharlieBackend.Business/Services/HomeworkStudentService.cs

```diff
@@ -0,0 +1,199 @@
+﻿using AutoMapper;
+using CharlieBackend.Business.Services.Interfaces;
+using CharlieBackend.Core.DTO.HomeworkStudent;
+using CharlieBackend.Core.Entities;
+using CharlieBackend.Core.Models.ResultModel;
+using CharlieBackend.Data.Repositories.Impl.Interfaces;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CharlieBackend.Business.Services
+{
+    public class HomeworkStudentService : IHomeworkStudentService
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
+        private readonly ILogger<HomeworkStudentService> _logger;
+
+        public HomeworkStudentService(IUnitOfWork unitOfWork, IMapper mapper, ILogger<HomeworkStudentService> logger)
+        {
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+            _logger = logger;
+        }
+
+        public async Task<Result<HomeworkStudentDto>> CreateHomeworkFromStudentAsync(HomeworkStudentRequestDto homeworkStudent, ClaimsPrincipal userContext)
```

Comment:
I think here u should remove ClaimsPrincipal and add CurrentUserService in future

Reply:
I already did it